### PR TITLE
feat(lidar_centerpoint): make lidar_centerpoint a shared library

### PR DESCRIPTION
**Pull request by yukke42**
## Description

- make lidar_centerpoint a shared library to be used from pointpainting
- most of core parameters of centerpoint can be loaded from ros param
  - some of parameters depend on the cuda implementation, so they're hard coded
- refactor the directory structure (the core processes haven't changed)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by tzhong518 — APPROVED** [review]

File: perception/lidar_centerpoint/CMakeLists.txt [hunk 1]
```diff
@@ -110,40 +110,49 @@ if(TRT_AVAIL AND CUDA_AVAIL AND CUDNN_AVAIL)
...
-    centerpoint_cuda_libraries
+    centerpoint_cuda_lib
+  )
+
+  target_include_directories(centerpoint_lib
+    PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      $<INSTALL_INTERFACE:include>
+    PRIVATE
+      ${CMAKE_CURRENT_SOURCE_DIR}/src
```
Comment (wep21) [inline perception/lidar_centerpoint/CMakeLists.txt:150]: Are there any header files in src directory?

Comment (yukke42) [reply perception/lidar_centerpoint/CMakeLists.txt:150]: > Are there any header files in src directory?
No, I'll fix it.

Comment (yukke42) [reply perception/lidar_centerpoint/CMakeLists.txt]: fixed at [280bf42](https://github.com/autowarefoundation/autoware.universe/pull/733/commits/280bf42f02e9203c28b90c61b9bcf2377db8aca4)

Comment (yukke42) [reply perception/lidar_centerpoint/CMakeLists.txt]: @wep21 Cloud you check if it's ok? :pray:

Comment (wep21) [reply perception/lidar_centerpoint/CMakeLists.txt]: @yukke42 looks good to me, thank you.

**Pull request merged**